src/sinks/datadog/metrics.rs

```diff
@@ -385,12 +436,59 @@ fn encode_events(events: Vec<Metric>, interval: i64, namespace: &str) -> Datadog
     DatadogRequest { series }
 }
 
+fn encode_distribution_events(
+    events: Vec<Metric>,
+    interval: i64,
+    namespace: &str,
+) -> DatadogRequest<DatadogDistributionMetric> {
+    debug!(message = "distribution", count = events.len());
+    let series = events
+        .into_iter()
+        .filter_map(|event| {
+            let fullname = encode_namespace(namespace, &event.name);
+            let ts = encode_timestamp(event.timestamp);
+            let tags = event.tags.clone().map(encode_tags);
+            match event.kind {
+                MetricKind::Incremental => match event.value {
+                    MetricValue::Distribution {
+                        values,
+                        sample_rates,
+                        statistic: StatisticKind::Summary,
+                    } => {
+                        let mut samples = Vec::new();
+                        for (v, c) in values.iter().zip(sample_rates.iter()) {
+                            for _ in 0..*c {
+                                samples.push(*v);
+                            }
+                        }
+
+                        if samples.is_empty() {
```

Comment:
Can this be moved above creating the `samples` array? If `values` is empty, then `samples` must also be empty.

Reply:
> If values is empty, then samples must also be empty.

Yes, but rate can be zero, so even a non empty values can result in empty samples.

Reply:
Fair enough, I was assuming the rates would be non-zero.